### ISSUE TITLE
Harmless error when saving defaults in magit-branch-config-popup: setting-constant nil

### ISSUE THREAD

**Issue by rdgreen** [magit/magit#2712]
I received the following error when attempting to save a change to the `branch.autoSetupRebase` setting in magit-branch-config-popup, which I access from the push command (P):

```
Debugger entered--Lisp error: (setting-constant nil)
  set-default(nil nil)
  customize-save-variable(nil nil)
  magit-popup-save-default-arguments(nil)
  call-interactively(magit-popup-save-default-arguments nil nil)
  command-execute(magit-popup-save-default-arguments)
```

However, the setting does in fact get saved, so I think it's complaining about another setting but it does still work.

**Comment by npostavs** [comment]
> the setting does in fact get saved,

The setting is saved immediately when you toggle it. This is the case for all git configuration variables. `magit-popup-save-default-arguments` only saves switches and options, since magit-branch-config-popup doesn't have any of those, this command is useless there.

We could trigger `user-error` when trying to use them, or remove them from such popups altogether?

**Comment by kyleam** [comment]
> We could trigger user-error when trying to use them, or remove them
> from such popups altogether?

I'd vote for making `magit-popup-save-default-arguments` and
`magit-popup-set-default-arguments` trigger a user error.

**Comment by npostavs** [comment]
Should be fixed by #2716.
